diff --git a/Makefile b/Makefile
index 8e47e3a..7123d14 100644
--- a/Makefile
+++ b/Makefile
@@ -325,8 +325,8 @@ generate: $(JAVASCRIPT_GEN)
 
 # --- Library ---
 
-MUPDF_LIB = $(OUT)/libmupdf.a
-THIRD_LIB = $(OUT)/libmupdfthird.a
+MUPDF_LIB = $(OUT)/libmupdf.so
+THIRD_LIB =
 THREAD_LIB = $(OUT)/libmuthreads.a
 PKCS7_LIB = $(OUT)/libmupkcs7.a
 
@@ -355,12 +355,13 @@ THIRD_OBJ := \
 	$(ZLIB_OBJ) \
 	$(LCMS2_OBJ)
 
-$(MUPDF_LIB) : $(MUPDF_OBJ)
+$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_OBJ)
+	$(LINK_CMD) -shared -Wl,-soname -Wl,libmupdf.so -Wl,--no-undefined
 $(THIRD_LIB) : $(THIRD_OBJ)
 $(THREAD_LIB) : $(THREAD_OBJ)
 $(PKCS7_LIB) : $(PKCS7_OBJ)
 
-INSTALL_LIBS := $(MUPDF_LIB) $(THIRD_LIB)
+INSTALL_LIBS := $(MUPDF_LIB)
 
 # --- Tools and Apps ---
 
