diff --git a/tensorflow/contrib/mpi_collectives/BUILD b/tensorflow/contrib/mpi_collectives/BUILD
index 9f9802b..a75a4ad 100644
--- a/tensorflow/contrib/mpi_collectives/BUILD
+++ b/tensorflow/contrib/mpi_collectives/BUILD
@@ -53,6 +53,7 @@ tf_custom_op_library(
         ":mpi_defines",
         ":mpi_message_proto_cc",
         "//third_party/mpi",
+        "//tensorflow/stream_executor",
     ],
 )
 
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index 51ef323..18c4b24 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -1279,7 +1279,6 @@ def tf_custom_op_library(name, srcs=[], gpu_srcs=[], deps=[], linkopts=[]):
       deps=deps + if_cuda(cuda_deps),
       disallowed_deps=[
           clean_dep("//tensorflow/core:framework"),
-          clean_dep("//tensorflow/core:lib")
       ])
   tf_cc_shared_object(
       name=name,
