diff --git a/tensorboard/pip_package/build_pip_package.sh b/tensorboard/pip_package/build_pip_package.sh
index 477e91c..363dd10 100755
--- a/tensorboard/pip_package/build_pip_package.sh
+++ b/tensorboard/pip_package/build_pip_package.sh
@@ -32,11 +32,6 @@ function main() {
   TMPVENVDIR="${TMPDIR}/venv"
   RUNFILES="bazel-bin/tensorboard/pip_package/build_pip_package.runfiles/org_tensorflow_tensorboard"
 
-  # Check that virtualenv is installed.
-  if [[ -z "$(which virtualenv)" ]]; then
-    die "ERROR: virtualenv is required, but does not appear to be installed."
-  fi
-
   echo $(date) : "=== Using tmpdir: ${TMPDIR}"
 
   bazel build //tensorboard/pip_package:build_pip_package
@@ -53,25 +48,11 @@ function main() {
   pushd ${TMPDIR} >/dev/null
   rm -f MANIFEST
 
-  echo $(date) : "=== Activating virtualev ==="
-  virtualenv "${TMPVENVDIR}"
-  export VIRTUAL_ENV="${TMPVENVDIR}"
-  export PATH="${TMPVENVDIR}/bin:${PATH}"
-  unset PYTHON_HOME
-
-  # Require wheel for bdist_wheel command, and setuptools 36.2.0+ so that
-  # env markers are handled (https://github.com/pypa/setuptools/pull/1081)
-  pip install -U wheel setuptools>=36.2.0
-
-  echo $(date) : "=== Building python2 wheel in $PWD"
-  python setup.py bdist_wheel --python-tag py2 >/dev/null
-  echo $(date) : "=== Building python3 wheel in $PWD"
-  python setup.py bdist_wheel --python-tag py3 >/dev/null
   mkdir -p ${DEST}
-  cp dist/* ${DEST}
+  cp -r * ${DEST}
   popd >/dev/null
   rm -rf ${TMPDIR}
-  echo $(date) : "=== Output wheel files are in: ${DEST}"
+  echo $(date) : "=== Source files are in: ${DEST}"
 }
 
 main "$@"
