diff --git a/tensorflow/tools/pip_package/build_pip_package.sh b/tensorflow/tools/pip_package/build_pip_package.sh
index dc31e4c..3c57c36 100755
--- a/tensorflow/tools/pip_package/build_pip_package.sh
+++ b/tensorflow/tools/pip_package/build_pip_package.sh
@@ -158,19 +158,14 @@ function main() {
   cp tensorflow/tools/pip_package/README ${TMPDIR}
   cp tensorflow/tools/pip_package/setup.py ${TMPDIR}
 
-  # Before we leave the top-level directory, make sure we know how to
-  # call python.
-  source tools/python_bin_path.sh
-
   pushd ${TMPDIR}
   rm -f MANIFEST
-  echo $(date) : "=== Building wheel"
-  "${PYTHON_BIN_PATH:-python}" setup.py bdist_wheel ${PKG_NAME_FLAG} >/dev/null
+  # We don't build wheel. We only copy the source files to the destination directory.
   mkdir -p ${DEST}
-  cp dist/* ${DEST}
+  cp -r * ${DEST}
   popd
   rm -rf ${TMPDIR}
-  echo $(date) : "=== Output wheel file is in: ${DEST}"
+  echo $(date) : "=== The Python package file is in: ${DEST}"
 }
 
 main "$@"
