diff --git a/src/config.mak b/src/config.mak
index e4558bb..d3b2e7f 100644
--- a/src/config.mak
+++ b/src/config.mak
@@ -16,7 +16,7 @@
 # The name of the Lua package (it could be "lua", "lua52", or "lua5.2")
 # Lua 5.3 works as well.
 #
-LUA_PACKAGE   ?= lua5.3
+LUA_PACKAGE   ?= lua
 #
 ZLIB_CFLAGS   ?=
 ZLIB_LIBS     ?= -lz
diff --git a/src/ipe/Makefile b/src/ipe/Makefile
index bd8fd58..9a93ae2 100644
--- a/src/ipe/Makefile
+++ b/src/ipe/Makefile
@@ -17,7 +17,7 @@ ifdef IPEBUNDLE
 CPPFLAGS += -DIPEBUNDLE
 else
 CPPFLAGS += -DIPELETDIR=\"$(IPELETDIR)\"
-CPPFLAGS += -DIPELUADIR=\"$(IPELUADIR)\"
+CPPFLAGS += -DIPELUADIR=\"/usr/share/ipe/$(IPEVERS)/lua\"
 CPPFLAGS += -DIPEICONDIR=\"$(IPEICONDIR)\"
 CPPFLAGS += -DIPEDOCDIR=\"$(IPEDOCDIR)\"
 CPPFLAGS += -DIPESTYLEDIR=\"$(IPESTYLEDIR)\"
