diff --git a/torch/CMakeLists.txt b/torch/CMakeLists.txt
index 9c4a018..b54037d 100644
--- a/torch/CMakeLists.txt
+++ b/torch/CMakeLists.txt
@@ -370,7 +370,7 @@ if(USE_CUDA)
     set_target_properties(torch PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
   else()
     set(TORCH_CUDA_LIBRARIES
-      ${CUDA_TOOLKIT_ROOT_DIR}/lib64/libnvToolsExt.so
+      /usr/lib64/libnvToolsExt.so
       ${CUDA_LIBRARIES})
   endif()
 
@@ -637,7 +637,7 @@ if (BUILD_PYTHON)
     elseif(APPLE)
       list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib/libnvToolsExt.dylib)
     else()
-      list(APPEND TORCH_PYTHON_LINK_LIBRARIES ${CUDA_TOOLKIT_ROOT_DIR}/lib64/libnvToolsExt.so)
+      list(APPEND TORCH_PYTHON_LINK_LIBRARIES /usr/lib64/libnvToolsExt.so)
     endif()
   endif()
 
