diff --git a/data/gnome/latexdraw.desktop b/data/gnome/latexdraw.desktop
index e3a47e7..c83b6c7 100644
--- a/data/gnome/latexdraw.desktop
+++ b/data/gnome/latexdraw.desktop
@@ -9,7 +9,7 @@ Categories=Graphics;VectorGraphics;LaTeX;
 MimeType=image/svg+xml;
 Exec=latexdraw %F
 TryExec=latexdraw
-Icon=/usr/share/latexdraw/images/app/latexdraw.png
+Icon=latexdraw
 Terminal=false
 StartupNotify=true
 X-Desktop-File-Install-Version=3.2
