diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddb229983..ac6de7c82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -251,7 +251,7 @@ if (BUILD_SHARED_LIBS)
   install(FILES
       ${PROJECT_BINARY_DIR}/Caffe2ConfigVersion.cmake
       ${PROJECT_BINARY_DIR}/Caffe2Config.cmake
-      DESTINATION share/cmake/Caffe2
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Caffe2
       COMPONENT dev)
   install(FILES
       ${PROJECT_SOURCE_DIR}/cmake/public/cuda.cmake
@@ -260,9 +260,9 @@ if (BUILD_SHARED_LIBS)
       ${PROJECT_SOURCE_DIR}/cmake/public/protobuf.cmake
       ${PROJECT_SOURCE_DIR}/cmake/public/threads.cmake
       ${PROJECT_SOURCE_DIR}/cmake/public/utils.cmake
-      DESTINATION share/cmake/Caffe2/public
+      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Caffe2/public
       COMPONENT dev)
-  install(EXPORT Caffe2Targets DESTINATION share/cmake/Caffe2
+  install(EXPORT Caffe2Targets DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Caffe2
       FILE Caffe2Targets.cmake
       COMPONENT dev)
 else()
diff --git a/caffe2/CMakeLists.txt b/caffe2/CMakeLists.txt
index 3225e9e53..b2bf854bc 100644
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -72,10 +72,10 @@ configure_file(
 
 # Installing the header files
 install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-        DESTINATION include
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
         FILES_MATCHING PATTERN "*.h")
 install(FILES ${PROJECT_BINARY_DIR}/caffe2/core/macros.h
-        DESTINATION include/caffe2/core)
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/caffe2/core)
 
 
 # ---[ List of libraries to link with
@@ -128,7 +128,7 @@ target_compile_options(caffe2 INTERFACE "-std=c++11")
 target_compile_options(caffe2 PRIVATE "-DCAFFE2_BUILD_MAIN_LIB")
 # Use -O2 for release builds (-O3 doesn't improve perf, and -Os results in perf regression)
 target_compile_options(caffe2 PRIVATE "$<$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebInfo>>:-O2>")
-install(TARGETS caffe2 EXPORT Caffe2Targets DESTINATION lib)
+install(TARGETS caffe2 EXPORT Caffe2Targets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 caffe2_interface_library(caffe2 caffe2_library)
 list(APPEND Caffe2_MAIN_LIBS caffe2_library)
 
@@ -153,7 +153,7 @@ if(USE_CUDA)
       caffe2_gpu PRIVATE ${Caffe2_CUDA_DEPENDENCY_LIBS})
   caffe2_interface_library(caffe2_gpu caffe2_gpu_library)
   list(APPEND Caffe2_MAIN_LIBS caffe2_gpu_library)
-  install(TARGETS caffe2_gpu EXPORT Caffe2Targets DESTINATION lib)
+  install(TARGETS caffe2_gpu EXPORT Caffe2Targets DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 # ---[ Test binaries.
@@ -171,6 +171,9 @@ if (BUILD_TEST)
     # result, we will explicitly link the test against the Caffe2 dependency
     # libs.
     target_link_libraries(${test_name} ${Caffe2_MAIN_LIBS} gtest_main)
+    if (USE_MPI)
+        target_link_libraries(${test_name} ${MPI_CXX_LIBRARIES})
+    endif()
     if (USE_CUDA)
       target_link_libraries(${test_name} ${Caffe2_CUDA_DEPENDENCY_LIBS})
     endif()
diff --git a/modules/detectron/CMakeLists.txt b/modules/detectron/CMakeLists.txt
index f18077b82..445bed80a 100644
--- a/modules/detectron/CMakeLists.txt
+++ b/modules/detectron/CMakeLists.txt
@@ -10,5 +10,5 @@ if (USE_CUDA)
       ${Detectron_GPU_SRCS})
 
   target_link_libraries(caffe2_detectron_ops_gpu caffe2_gpu)
-  install(TARGETS caffe2_detectron_ops_gpu DESTINATION lib)
+  install(TARGETS caffe2_detectron_ops_gpu DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
diff --git a/modules/module_test/CMakeLists.txt b/modules/module_test/CMakeLists.txt
index 3e9771417..9b8e6f809 100644
--- a/modules/module_test/CMakeLists.txt
+++ b/modules/module_test/CMakeLists.txt
@@ -11,5 +11,5 @@ if (BUILD_TEST)
       ${CMAKE_CURRENT_SOURCE_DIR}/module_test_dynamic.cc)
 
   target_link_libraries(caffe2_module_test_dynamic caffe2_library)
-  install(TARGETS caffe2_module_test_dynamic DESTINATION lib)
+  install(TARGETS caffe2_module_test_dynamic DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
diff --git a/modules/observers/CMakeLists.txt b/modules/observers/CMakeLists.txt
index 7bf872e37..e135cb7c0 100644
--- a/modules/observers/CMakeLists.txt
+++ b/modules/observers/CMakeLists.txt
@@ -16,7 +16,7 @@ add_library(caffe2_observers
     )
 target_link_libraries(caffe2_observers PUBLIC caffe2_library)
 target_include_directories(caffe2_observers PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/..)
-install(TARGETS caffe2_observers DESTINATION lib)
+install(TARGETS caffe2_observers DESTINATION ${CMAKE_INSTALL_LIBDIR})
 caffe2_interface_library(caffe2_observers caffe2_observers_library)
 
 if (CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
diff --git a/modules/rocksdb/CMakeLists.txt b/modules/rocksdb/CMakeLists.txt
index 242423e04..c2f0c387c 100644
--- a/modules/rocksdb/CMakeLists.txt
+++ b/modules/rocksdb/CMakeLists.txt
@@ -53,7 +53,7 @@ add_library(caffe2_rocksdb ${CMAKE_CURRENT_SOURCE_DIR}/rocksdb.cc)
 target_link_libraries(caffe2_rocksdb PUBLIC caffe2_library)
 target_link_libraries(caffe2_rocksdb PRIVATE ${RocksDB_LIBRARIES})
 target_include_directories(caffe2_rocksdb PRIVATE ${RocksDB_INCLUDE_DIR})
-install(TARGETS caffe2_rocksdb DESTINATION lib)
+install(TARGETS caffe2_rocksdb DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 # ---[ Last, Append the library to Caffe2_MAIN_LIBS, if we are building with
 # the main repo.
