diff --git a/configure.py b/configure.py
index 6bb1c5f..c74890f 100644
--- a/configure.py
+++ b/configure.py
@@ -1432,6 +1432,9 @@ def main():
   check_bazel_version('0.15.0')
 
   reset_tf_configure_bazelrc(args.workspace)
+  # Explicitly import tools/bazel.rc, this is needed for Bazel 0.19.0 or later
+  write_to_bazelrc('import %workspace%/tools/bazel.rc')
+
   cleanup_makefile()
   setup_python(environ_cp)
 
